engine/table/src/main/java/io/deephaven/engine/table/impl/sources/regioned/kernel/CharRegionBinarySearchKernel.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) 2016-2023 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.engine.table.impl.sources.regioned.kernel;
+
+import io.deephaven.api.SortColumn;
+import io.deephaven.chunk.WritableCharChunk;
+import io.deephaven.chunk.attributes.Any;
+import io.deephaven.engine.rowset.RowSet;
+import io.deephaven.engine.rowset.RowSetBuilderSequential;
+import io.deephaven.engine.rowset.RowSetFactory;
+import io.deephaven.engine.table.impl.sort.timsort.CharTimsortDescendingKernel;
+import io.deephaven.engine.table.impl.sort.timsort.CharTimsortKernel;
+import io.deephaven.engine.table.impl.sources.regioned.ColumnRegionChar;
+import io.deephaven.util.compare.CharComparisons;
+import io.deephaven.util.type.ArrayTypeUtils;
+import org.jetbrains.annotations.NotNull;
+
+public class CharRegionBinarySearchKernel {
+    public static RowSet binSearchMatch(
```

Comment:
Can you add comments to this function?  It's the sole user-facing function and some of parameters have misleading names.